cmd/lakectl/cmd/common_helpers.go

```diff
@@ -315,6 +317,17 @@ func MustParsePathURI(name, s string) *uri.URI {
 	return u
 }
 
+func MustGetContentType(flagSet *pflag.FlagSet, sourcePathname string) string {
+	contentType, err := flagSet.GetString("content-type")
+	if err != nil {
+		DieErr(err)
+	}
+	if len(contentType) != 0 {
+		return contentType
+	}
+	return mime.TypeByExtension(filepath.Ext(sourcePathname))
```

Comment:
*tl;dr:* Auto content-type detection from extension _must_ be controllable by the user, and it _should probably_ not happen by default.

I know that this implements the requirement of the issue.

This automatic functionality is unexpected and difficult to predict, for multiple reasons:
* It happens automatically.
* It does not look at content.  But most applications _do_ look at content, so this is vulnerable in various ways.  Even Windows has stopped auto-associating content types with file extensions, AFAIK.
* [It does different things on different systems](https://pkg.go.dev/mime#TypeByExtension), according to their `/etc/mime.types`.  And on Windows it will do even less.